routes.go

```diff
@@ -23,119 +23,132 @@ import (
-			return alice.Chain{}, err
-		}
-
-		requestDuration, err := in.RequestDuration.CurryWith(curryLabel)
-		if err != nil {
-			return alice.Chain{}, err
-		}
-
-		requestsInFlight, err := in.RequestsInFlight.CurryWith(curryLabel)
-		if err != nil {
-			return alice.Chain{}, err
-		}
-
-		return alice.New(
-			xmetricshttp.HandlerCounter{
-				Metric:   xmetrics.LabelledCounterVec{CounterVec: requestCount},
-				Labeller: serverLabellers,
-			}.Then,
-			xmetricshttp.HandlerDuration{
-				Metric:   xmetrics.LabelledObserverVec{ObserverVec: requestDuration},
-				Labeller: serverLabellers,
-			}.Then,
-			xmetricshttp.HandlerInFlight{
-				Metric: xmetrics.LabelledGaugeVec{GaugeVec: requestsInFlight},
-			}.Then,
-		), nil
-	})
+type SubRouterOut struct {
+	fx.Out
+	Router *mux.Router `name:"server_primary_accounts"`
 }
 
-type PrimaryRouter struct {
+type PrimaryMMIn struct {
 	fx.In
-	Router    *mux.Router `name:"servers.primary"`
-	AuthChain alice.Chain `name:"primary_auth_chain"`
+	Primary alice.Chain `name:"middleware_primary_metrics"`
 }
 
-type PrimaryRoutes struct {
+type HealthMMIn struct {
 	fx.In
-	Set    store.Handler `name:"set_handler"`
-	Delete store.Handler `name:"delete_handler"`
-	Get    store.Handler `name:"get_handler"`
-	GetAll store.Handler `name:"get_all_handler"`
+	Health alice.Chain `name:"middleware_health_metrics"`
+}
+
+type MetricMiddlewareOut struct {
+	fx.Out
+	Primary alice.Chain `name:"middleware_primary_metrics"`
+	Health  alice.Chain `name:"middleware_health_metrics"`
 }
 
-func BuildPrimaryRoutes(router PrimaryRouter, routes PrimaryRoutes) {
-	router.Router.Use(router.AuthChain.Then)
-	itemPathFormat := "/%s/store/{bucket}/{id}"
+func provideServers() fx.Option {
+	return fx.Options(
+		arrangehttp.Server{
+			Name: "server_primary",
+			Key:  "servers.primary",
+			Inject: arrange.Inject{
+				PrimaryMMIn{},
+			},
+		}.Provide(),
+		fx.Provide(
+			metricMiddleware,
+		),
+		arrangehttp.Server{
+			Name: "server_health",
+			Key:  "servers.health",
+			Inject: arrange.Inject{
+				HealthMMIn{},
+			},
+			Invoke: arrange.Invoke{
+				func(r *mux.Router) {
+					r.Handle("/health", httpaux.ConstantHandler{
+						StatusCode: http.StatusOK,
+					}).Methods("GET")
+				},
+			},
+		}.Provide(),
+		arrangehttp.Server{
+			Name: "server_metrics",
+			Key:  "servers.metrics",
+		}.Provide(),
+
+		fx.Invoke(
+			handlePrimaryEndpoint,
+			handleMetricEndpoint,
+		),
+	)
+}
 
-	if routes.Set != nil {
-		router.Router.Handle(fmt.Sprintf(itemPathFormat, apiBase), routes.Set).Methods(http.MethodPut)
+func handlePrimaryEndpoint(in PrimaryRouterIn) {
+	options := []otelmux.Option{
+		otelmux.WithTracerProvider(in.Tracing.TracerProvider()),
+		otelmux.WithPropagators(in.Tracing.Propagator()),
 	}
-	if routes.Get != nil {
-		router.Router.Handle(fmt.Sprintf(itemPathFormat, apiBase), routes.Get).Methods(http.MethodGet)
+	in.Router.Use(
+		in.AuthChain.Then,
+		otelmux.Middleware("servers_primary", options...),
```

Comment:
I just made it match the name used for uber fx wiring of the server...With the wiring, I think `servers_primary` is more descriptive than just `primary` by itself.  I'm not sure if this needs to match or not. :shrug:

Reply:
yeah, `server.primary`, `servers.primary` or `servers_primary` work for me 👍🏼 

This will show up in the trace details when looking at them in tracing dashboards so we probably want to make them all the same across services